src/openforms/submissions/models/submission.py

```diff
@@ -339,6 +339,20 @@ class Meta:
                 ),
                 name="registration_status_consistency_check",
             ),
+            models.CheckConstraint(
+                check=(
+                    models.Q(finalised_registration_backend_key="")
+                    | (
+                        ~models.Q(finalised_registration_backend_key="")
+                        & models.Q(completed_on__isnull=False)
+                    )
```

Comment:
Doesn't the ` | ` part takes into account the fact that `~models.Q(finalised_registration_backend_key="")`? So maybe it can be simplified as:

```suggestion
                    models.Q(finalised_registration_backend_key="")
                    | (
                        models.Q(completed_on__isnull=False)
                    )
```

Reply:
yes, I had that initially but then opted for a more explicit variant to really illustrate the two different cases (by including the common factor between the two cases)

I'd expec the ORM/database to optimize this anyway